homeassistant/components/generic/camera.py

```diff
@@ -77,10 +84,18 @@ def __init__(self, hass, device_info):
         self.hass = hass
         self._authentication = device_info.get(CONF_AUTHENTICATION)
         self._name = device_info.get(CONF_NAME)
-        self._still_image_url = device_info[CONF_STILL_IMAGE_URL]
-        self._stream_source = device_info.get(CONF_STREAM_SOURCE)
+        strg = device_info[CONF_STILL_IMAGE_URL]
+        if isinstance(strg, str) and strg != "":
+            self._still_image_url = cv.template(device_info[CONF_STILL_IMAGE_URL])
+        else:
+            self._still_image_url = device_info[CONF_STILL_IMAGE_URL]
         self._still_image_url.hass = hass
-        if self._stream_source is not None:
+        strg = device_info.get(CONF_STREAM_SOURCE)
```

Comment:
Instead of a random variable here, go ahead and set `self._stream_source` and check that it's not already a template in the single `if` condition below, eliminating the `else`.